[RFC] multiple things on ath79 and ar71xx

p-wassi opened the pull request:
Multiple commits to discuss, everything's RFC!

- fix TL-WR741ND-v4 switch port order:
Just installed ath79 on that device and noticed, the ports in `board.json` were not as expected.
Set port order to the same as in WR740N-v4 (also the LED trigger bitmasks for these two devices were already equivalent, i.e. the switch layout is the same)

- ath79: rename TL-WR740ND-v4 to TL-WR740N-v4
On ar71xx, the board was called tl-wr740n-v4. Versions v1 and v3 are also called "tl-wr740n-v_".
If you google for the device, I can only find **offical** tp-link.com sites for WR740N, which is not
the case for e.g. WR841N/ND. For this device(s), tp-link.com actually has different product pages.
The OpenWrt wiki at https://oldwiki.archive.openwrt.org/toh/tp-link/tl-wr740n doesn't even mention WR740ND. (I know, it's the page tl-wr740n, but there's no page tl-wr740nd)

- While we're at it:
Compare ar71xx and ath79 for wr841:
`grep wr841 target/linux/ar71xx/base-files/etc/board.d/02_network`:
```
	tl-wr841n-v11|\
	tl-wr841n-v9|\
	tl-wr841n-v8|\
	tl-wr841n-v1|\
	tl-wr841n-v7)
```
`grep wr841 target/linux/ath79/base-files/etc/board.d/02_network`:
```
	tplink,tl-wr841-v7|\
	tplink,tl-wr841-v9|\
	tplink,tl-wr841-v11|\
	tplink,tl-wr841-v5|\
	tplink,tl-wr841-v8|\
```

- fix TL-WR741ND-v4 switch port order on ar71xx too
I could not believe, that this has been (undetected) in ar71xx for so many years,
I 'downgraded' to ar71xx and installed Luci just to confirm.
The LED trigger pattern on ar71xx already reflected the correct switch layout.

- While doing this on ar71xx, another thing 'popped up':
the MR3220-v2 has exactly the same LED trigger pattern for the switch as 741ND-v4.
Shouldn't it then also have the same swconfig?

- Finally, during `make menuconfig` I came across the 'issue' that TP-Link
devices are (not) in alphabetical order. Some are called 'TP-LINK ...', others 'TP-Link ...'.
Remove the inconsistencies and use the vendors name 'TP-Link'

- change ledtrig on GL.iNet AR150
Due to the reasons discussed in #1704 :
On ar71xx we could have netdev triggers for devices that had a single **switch**port routed to a RJ45 jack. Here, we have to use the switch trigger instead.
Does this change in behaviour (ar71xx vs ath79) have relevant effects on the hotplug system?
Back then, `eth_` really went up and down in kernel, probably triggering a DHCP renew for WAN, etc. 



I could only run-test on TL-WR741ND-v4 and GL-AR150, this is RFC!
// EDIT:
BTW: WR741ND-v4 also needs #1700 (same chip)

p-wassi committed:
ath79: fix TL-WR741ND-v4 switch port order

Fix the switch port order for proper display on high
level interfaces.

Signed-off-by: Paul Wassi <p.wassi@gmx.at>

p-wassi committed:
ath79: rename TL-WR740ND-v4 to TL-WR740N-v4

Give the device the same name it had in ar71xx.

Signed-off-by: Paul Wassi <p.wassi@gmx.at>

p-wassi committed:
ar71xx: fix TL-WR741ND-v4 switch port order

Fix the switch port order for proper display on high
level interfaces.

Signed-off-by: Paul Wassi <p.wassi@gmx.at>

p-wassi committed:
ar71xx: fix TL-MR3220-v2 switch port order

Fix the switch port order for proper display on high
level interfaces.

Signed-off-by: Paul Wassi <p.wassi@gmx.at>

p-wassi committed:
ath79: rename TP-LINK to TP-Link

Remove inconsistencies in the vendor's name.

Signed-off-by: Paul Wassi <p.wassi@gmx.at>

p-wassi committed:
ath79: change ledtrig on GL.iNet AR150

Change the ledtrig for LAN from netdev to switch.
Although eth1 comes out of the device at a single port,
this port is a switch-port and therefore the LED
must be triggered by that.

Signed-off-by: Paul Wassi <p.wassi@gmx.at>

981213 commented:
> I could not believe, that this has been (undetected) in ar71xx for so many years,

Here are my guessed reasons:
1. I think this isn't considered a 'problem' because the original code aren't setting the wrong port numbers, they just don't set the optional port numbers. As long as the port are in the correct group (lan/wan), the port ordering itself isn't a critical problem.
2. From my personal experience I did't know those port numbers can be specified for a long time. Even after I know this I tend to use the existing working entries instead of adding new ones when adding new device support.

> the MR3220-v2 has exactly the same LED trigger pattern for the switch as 741ND-v4.
> Shouldn't it then also have the same swconfig?

TP-Link likes to use the same board design for many routers with different cases, and these different cases can have different port numbering (lan1/2/3/4/wan vs lan4/3/2/1/wan) while using exactly the same firmware(e.g. lan port numbers between TL-WR941N v4 and TL-WR940N v1 are reversed). In ar71xx these devices shares the same piece of machine setup code and their LED names can be wrong. I suggest not to change this without confirming the port numbering on the case from actual device pictures.

p-wassi committed:
ath79: fix port order on TL-WR841ND-v7

The switch ports are seen one to one on the case.
Also remove unneeded secondary port numbers in this
case statement.

Signed-off-by: Paul Wassi <p.wassi@gmx.at>

p-wassi commented:
New commit:
- fix port order on TL-WR841ND-v7
The port numbers for TL-WR841ND-v7 are not correct on ath79.
However they are correct on ar71xx.
While we're at this case statement remove redundant secondary port numbers.

Run tested on WR841ND-v7

p-wassi committed:
ath79: proper indentation in image/common-tp-link.mk

Add two spaces for proper indentation in image/common-tp-link.mk

Signed-off-by: Paul Wassi <p.wassi@gmx.at>

chunkeey commented:
Merged. Thanks. (I added some more TP-LINK -> TP-Link changes that happend in the mean time)
